Update page attributes after edit

marcfrederick opened the pull request:
As discussed in #354, this change updates the page attributes without extra API calls after each (successful) call to `Page._edit` and its upstream functions (`Page.save`, `Page.edit`, `Page.append`, `Page.prepend`, and `Page.touch`). This is more or less what I suggested in [this comment](https://github.com/mwclient/mwclient/issues/354#issuecomment-2317762162).

@AdamWill [you mentioned](https://github.com/mwclient/mwclient/issues/354#issuecomment-2318077642) having an alternative patch lying around. I've checked the "Allow edits and access to secrets by maintainers" option, so you should be able to push to this branch if you have any changes/improvements over this.

resolves #354, resolves #63

AdamWill commented:
So compared to my patch, this handles the `edit` path much more extensively (nice job), but I also figured it made sense to update status in `move`:

        result = self.site.post('move', ('from', self.name), to=new_title,
                                token=self.get_token('move'), reason=reason, **data)
        if 'redirectcreated' not in result:
            # no redirect, so this page no longer exists
            self.exists = False
        return result['move']

and `delete`:

        result = self.site.post('delete', title=self.name,
                                token=self.get_token('delete'),
                                reason=reason, **data)
        # if the POST did not raise an exception, we don't exist
        self.exists = False
        return result['delete']

wdyt about those?

marcfrederick commented:
Yeah, makes sense. I’ve updated the PR with those changes and added some tests. I’ve also changed `Page.move` to set `Page.exists = False` when no redirect is created.

AdamWill commented:
@RheingoldRiver wdyt? I think you're entirely correct that the pages-as-persistent-instances design is inherently a bit wrong-headed and thus this can *always* potentially be a lie, but given we're stuck maintaining a mature library with said wrong-headed design, making it *less likely* to be a lie is overall still a win.

AdamWill reviewed (approved):
LGTM.

RheingoldRiver commented:
I think this should be okay, it's a pretty conservative change, and I can't imagine any sane situation in which it would break someone's code. LGTM
